### DISCUSSION TITLE
Download failed

### DISCUSSION THREAD

**Discussion by exedotp** [aclap-dev/video-downloadhelper#604] [Help and Q&A]
### Your question

{
  "version": "9.0.1.18",
  "target": "google",
  "channel": "stable",
  "page_url": "https://www.jeyadra.com/s/courses/65dc1691e4b0336e981247ce/take",
  "error": {
    "details": "SideDownload error: : [hls @ 00000174e2e1a500] Error when loading first segment 'https://dnqe0v482iyb1.cloudfront.net/spees/w/o/61af88470cf272041d7a8088/v/664d7d90ba3bad3c59903032/u/66502c206c9a19740076b5ef/t/795ca4858977cddd89707652bfda4f90/p/assets/videos/61af88470cf272041d7a8088/2024/05/22/664d7d90ba3bad3c59903032/hls_2M_000.ts'\r\n[in#0 @ 00000174e1448c40] Error opening input: Invalid data found when processing input\r\nError opening input file https://dnqe0v482iyb1.cloudfront.net/spees/w/o/61af88470cf272041d7a8088/v/664d7d90ba3bad3c59903032/u/66502c206c9a19740076b5ef/t/795ca4858977cddd89707652bfda4f90/p/assets/videos/61af88470cf272041d7a8088/2024/05/22/664d7d90ba3bad3c59903032/hls_2M_.m3u8.\r\nError opening input files: Invalid data found when processing input\r\n",
    "downloadable_id": "downloadable_8061378773672156",
    "error": "coapp_failure",
    "id": "coapp_failure_9530ada0-7196-4e87-8f68-c6e92e49dae1"
  }
}

### Addon & CoApp version and details

version: 9.0.1.18
target: google
channel: stable
lang: en-GB
coapp: {"found":true,"new_version":false,"path":"C:\\Program Files\\DownloadHelper CoApp\\vdhcoapp.exe","version":"2.0.19"}
license: {"unset":true}
platform: x86-64 win
UA: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/124.0.0.0 Safari/537.36
{
    "used_sidebar_button": true,
    "download_directory": "C:\\Users\\sunny\\dwhelper",
    "view_options": {
        "all_tabs": false,
        "hide_downloaded": false,
        "low_quality": false,
        "show_button_clean": true,
        "show_button_clean_all": false,
        "show_button_convert_local": false,
        "sort_by_status": true,
        "sort_reverse": false
    },
    "use_sidebar": true,
    "open_count_store": 1
}

**Comment by paulrouget** [comment]
Thank you for the report. We'll look into this!

Note: ff failure: "Invalid data found when processing input".

**Reply by GrampaWildWilly** [reply]
I looked at this.  The page_url is a login page, so that's a dead end.  The manifest, however, may contain a clue.  It's not a master manifest but rather a stream manifest.  It starts with this:

#EXT-X-KEY:METHOD=AES-128,URI="k/timestamp",IV=0x496daa1c6914000e408c65cead91fc29

I have a feeling there's some monkey business going on here with the decrypting.  My attempt to download this via ffmpeg using this stream manifest just quits after reading 4 chunks of the stream.  The 4 reads appear to be fine.  There's no errors.  But after the 4th read, it generates an error about the FIRST chunk.  I would have expected it to generate an error when it read the first chunk instead of running on through 3 more chunks.  In any case, I suspect it's not decrypting the blocks correctly.  Now why would that be?  Beats me.
